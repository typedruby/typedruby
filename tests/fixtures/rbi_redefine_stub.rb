# @typedruby
require_relative "_lib/has_stub.rb"

# Redefining a method already defined in a .rbi file is illegal.
class HasStub
  def hi
  end
end
