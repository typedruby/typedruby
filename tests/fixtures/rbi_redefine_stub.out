error: Can't redefine a method that was defined as a stub.

        @ __ROOT__/tests/fixtures/rbi_redefine_stub.rb:6
          ---v
      6 | |  def hi
      7 | |  end
          -----^ re-defined here
        @ __ROOT__/tests/fixtures/_lib/has_stub.rbi:2
      2 |    def hi => nil
                       ^^^ stub defined here
